templates/worker/00-worker/baremetal/files/baremetal-non-virtual-ip.yaml

```diff
@@ -4,29 +4,76 @@ path: "/usr/local/bin/non_virtual_ip"
 contents:
   inline: |
     #!/usr/libexec/platform-python
-    import collections
+    import itertools
     import socket
     import struct
     import subprocess
     import sys
+    from typing import Callable, Iterator, List, Optional, Tuple, Type, TypeVar
 
 
     class SubnetNotFoundException(Exception):
         """
         Exception raised when no subnet in the systems ifaces is on the VIP subnet
         """
-        pass
 
 
-    Address = collections.namedtuple('Address', 'index name family cidr scope')
+    TA = TypeVar('TA', bound='Address')
+
+    class Address:
+        def __init__(self, cidr: str, name: str, family: str, index: int = -1, scope: str = '') -> None:
+            self.index = index
+            self.name = name
+            self.family = family
+            self.cidr = cidr
+            self.scope = scope
+
+        @classmethod
+        def from_line(cls: Type[TA], line: str) -> TA:
+            items = line.split()
+            return cls(index=int(items[0][:-1]),
+                       name=items[1],
+                       family=items[2],
+                       cidr=items[3],
+                       scope=items[5])
+
+        def __str__(self) -> str:
+            return f'{self.__class__.__name__}({self.cidr}, dev={self.name})'
+
+
+    TR = TypeVar('TR', bound='V6Route')
+
+    class V6Route:
+        def __init__(self, destination: str, dev: str, proto: str, metric: int, pref: str, via: Optional[str] = None) -> None:
+            self.destination: str = destination
+            if via is not None:
+                self.via: str = via
+            self.dev: str = dev
+            self.proto: str = proto
+            self.metric: int = metric
+            self.pref: str = pref
+
+        @classmethod
+        def from_line(cls: Type[TR], line: str) -> TR:
+            items = line.split()
+            dest = items[0]
+            if dest == 'default':
+                dest = '::/0'
+            attrs = dict(itertools.zip_longest(*[iter(items[1:])]*2, fillvalue=None))
+            attrs['destination'] = dest
```

Comment:
proto isn't always set, so I think we need `attrs.setdefault('proto', '')` or similar here, to avoid a possible traceback with a line like `fd01::/48 via fd01:0:0:4::1 dev k8s-worker-0.os metric 1024 pref medium`

Reply:
Thanks @hardys. Will update